django/db/models/query.py

```diff
@@ -1683,10 +1683,7 @@ def __init__(self, klass_info, select, db):
             reorder_map.sort()
             self.init_list = [v[1] for v in reorder_map]
             pos_list = [row_pos for _, _, row_pos in reorder_map]
-
-            def reorder_for_init(row):
-                return [row[row_pos] for row_pos in pos_list]
-            self.reorder_for_init = reorder_for_init
+            self.reorder_for_init = operator.itemgetter(*pos_list)
```

Comment:
`pos_list` could be changed to be a generator now that it is only unpacked into `itemgetter`:

```diff
-pos_list = [row_pos for _, _, row_pos in reorder_map]
-self.reorder_for_init = operator.itemgetter(*pos_list)
+self.reorder_for_init = operator.itemgetter(*(v[2] for v in reorder_map))
```